File: frontend/src/app/models/note.ts
import { Tag } from "./tag";

export class Note {

  id: number = 0;
  title!: string;
  content!: string;
  finished!: boolean;
  tag: Tag | null;

  constructor(id:number = 0 ,title: string = '', content: string = '', finished: boolean = false, tag: Tag | null = null) {
    this.id = id;
    this.title = title;
    this.content = content;
    this.finished = finished;
    this.tag = tag;
  }

}

File: backend/src/main/java/com/app/notes/persistence/entity/Note.java
package com.app.notes.persistence.entity;
import lombok.Data;
import javax.persistence.*;
import javax.validation.constraints.NotBlank;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.LocalDateTime;


@Data
@Entity
@Table(name = "note")
public class Note {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotBlank
    private String title;

    @NotBlank
    private String content;
    private LocalDateTime createdDate;
    private boolean finished; 

    @ManyToOne(cascade = CascadeType.ALL)
    @JsonIgnoreProperties(value = {"applications", "hibernateLazyInitializer"})
    private Tag tag;

}